include/spatio_temporal_voxel_layer/spatio_temporal_voxel_grid.hpp

```diff
@@ -153,7 +156,7 @@ class SpatioTemporalVoxelGrid
   int                             _decay_model;
   double                          _background_value, _voxel_size, _voxel_decay;
   bool                            _pub_voxels;
-  pcl::PointCloud<pcl::PointXYZ>::Ptr     _pc;
+  std::list<geometry_msgs::Point32>*   _grid_points;
```

Comment:
I'm not sure how I feel about this. However if we're going to move to an STL container, it should probably be a `unordered_map` or `map` (unordered is slightly faster for iterating) and have the name as a `typedef` to abstract it

Reply:
I used the same approach as in the PR you pointed: https://github.com/ros-planning/navigation/pull/755/files
I did not change to unordered_map yet, but I have no problem with switching (side note, it is only used for visualization)

Reply:
Coming back on this point, I don't understand why you want to use `map` or `unordered_map`  to store `geometry_msgs::Point32` objects. Why using need associative containers here?
However, one can argue `vector` could be faster than `list` for this use case. https://baptiste-wicht.com/posts/2012/12/cpp-benchmark-vector-list-deque.html

Reply:
Lets change this to an `unordered_set` that gives us faster random access.